### PR TITLE
crimson, cls: fix the inability to print logs from plugins.

### PR DESCRIPTION
**Pull request by rzarzynski**
`cls_log()` of the interface between an OSD and a plugin (a Ceph Class) is implemented on top of the `dout` macros and shared between crimson and the classical OSD.

Unfortunately, when a plugin is hosted inside crimson, this causes the inability to send to any in-plugin generated message to the `seastar::logger` for the `objclass` subsystem.

This patch differtiates the implementation of `cls_log()`, and thus allow the `seastar::logger`-based implementation of `dout` to be used.

After the fix:

```
DEBUG 2022-03-07 14:12:14,124 [shard 0] osd - handling op call on object 1:424638ea:::image1:head
DEBUG 2022-03-07 14:12:14,124 [shard 0] osd - calling method rbd.create, num_read=0, num_write=0
DEBUG 2022-03-07 14:12:14,125 [shard 0] objclass - <cls> ../src/cls/rbd/cls_rbd.cc:787: create object_prefix=image1 size=0 order=22 features=0
DEBUG 2022-03-07 14:12:14,125 [shard 0] osd - handling op omap-get-vals-by-keys on object 1:424638ea:::image1:head
DEBUG 2022-03-07 14:12:14,125 [shard 0] osd - omap_get_vals_by_keys: object does not exist: 1:424638ea:::image1:head
```

Signed-off-by: Radoslaw Zarzynski <rzarzyns@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/osd/objclass.cc [hunk 1]
```diff
@@ -527,3 +531,23 @@ int cls_cxx_get_gathered_data(cls_method_context_t hctx, std::map<std::string, b
...
 }
+
+// although at first glance the implementation looks the same as in
+// the clasical OSD, it's different b/c of how the dout macro expand.
+int cls_log(int level, const char *format, ...)
+{
+   int size = 256;
+   va_list ap;
+   while (1) {
+     char buf[size];
```
[the remark this answers is not in the document]
Comment (rzarzynski) [reply src/crimson/osd/objclass.cc:542]: Yup, makes sense. Will push as a separate commit.

[the remark this answers is not in the document]
Comment (rzarzynski) [reply src/crimson/osd/objclass.cc]: Pushed.

**Review by athanatos — APPROVED** [review]

**Comment by rzarzynski** [comment]
jenkins retest this please

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
jenkins test make check

**Review by liu-chunmei — APPROVED** [review]

**Pull request merged**